### PR TITLE
Add new connector to support read access on AWS::Location::PlaceIndex resources

### PR DESCRIPTION
**Pull request by dylan8902**
### Description of changes

* Adds new connector make use of AWS Location service
* Updates guide to use more recent python version 3.7 patch that was installed earlier in developer guide

### Description of how you validated changes

Unit tests have been added

### Checklist

- [x] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [x] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/model/connector_profiles/profiles.json [hunk 1]
```diff
@@ -352,6 +352,47 @@
...
+                    "geo:CalculateRoute",
+                    "geo:GetGeofence",
+                    "geo:DescribePlaceIndex",
+                    "geo:SearchPlaceIndexForSuggestions",
+                    "geo:DescribeGeofenceCollection",
+                    "geo:SearchPlaceIndexForPosition",
+                    "geo:GetPlace",
+                    "geo:ListTrackerConsumers",
+                    "geo:DescribeRouteCalculator",
+                    "geo:GetMapTile"
```
Comment (hoffa) [inline samtranslator/model/connector_profiles/profiles.json:388]: So it's easier to read at a glance:

```suggestion
                    "geo:BatchGetDevicePosition",
                    "geo:CalculateRoute",
                    "geo:CalculateRouteMatrix",
                    "geo:DescribeGeofenceCollection",
                    "geo:DescribeMap",
                    "geo:DescribePlaceIndex",
                    "geo:DescribeRouteCalculator",
                    "geo:DescribeTracker",
                    "geo:GetDevicePosition",
                    "geo:GetDevicePositionHistory",
                    "geo:GetGeofence",
                    "geo:GetMapGlyphs",
                    "geo:GetMapSprites",
                    "geo:GetMapStyleDescriptor",
                    "geo:GetMapTile",
                    "geo:GetPlace",
                    "geo:ListDevicePositions",
                    "geo:ListGeofences",
                    "geo:ListTagsForResource",
                    "geo:ListTrackerConsumers",
                    "geo:SearchPlaceIndexForPosition",
                    "geo:SearchPlaceIndexForSuggestions",
                    "geo:SearchPlaceIndexForText"
```

File: tests/translator/input/connector_function_to_location.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        };
+
+  MyConnector:
+    Type: AWS::Serverless::Connector
+    Properties:
+      Source:
+        Id: MyFunction
+      Destination:
+        Arn: arn:aws:geo:us-east-1:123123123123:place-index/explore.place
+        Type: AWS::Location::PlaceIndex
```
Comment (hoffa) [inline tests/translator/input/connector_function_to_location.yaml:20]: Can you also add another connector that references an `Id` to a `AWS::Location::PlaceIndex` in the same template as well?

Comment (dylan8902) [reply tests/translator/input/connector_function_to_location.yaml:26]: Absolutely - done

**Comment by ssenchenko** [comment]
Thank you for your proposal @dylan8902 
Since it's a new profile, we need to get it through the review (including security). It will take some time.

File: .cfnlintrc.yaml [hunk 3]
```diff
@@ -47,6 +47,7 @@ ignore_templates:
   - tests/translator/output/**/connector_bucket_to_function.json
   - tests/translator/output/**/connector_dependson_replace.json
   - tests/translator/output/**/connector_esm_dependson.json
+  - tests/translator/output/**/connector_function_to_location.json
```
Comment (hoffa) [inline .cfnlintrc.yaml:50]: What does `cfn-lint` complain about? If there's an easy way to keep it happy, that would be preferred over ignoring the file outright.

Comment (dylan8902) [reply .cfnlintrc.yaml]: Nothing after running `make lint` so have removed the ignore

File: samtranslator/model/connector_profiles/profiles.json [hunk 4]
```diff
@@ -352,6 +352,47 @@
...
+                    "geo:GetMapStyleDescriptor",
+                    "geo:GetMapTile",
+                    "geo:GetPlace",
+                    "geo:ListDevicePositions",
+                    "geo:ListGeofences",
+                    "geo:ListTagsForResource",
+                    "geo:ListTrackerConsumers",
+                    "geo:SearchPlaceIndexForPosition",
+                    "geo:SearchPlaceIndexForSuggestions",
+                    "geo:SearchPlaceIndexForText"
```
Comment (GavinZZ) [inline samtranslator/model/connector_profiles/profiles.json:388]: Reading from [Actions defined by Amazon Location](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonlocation.html#amazonlocation-actions-as-permissions), a READ permission to a `AWS::Location::PlaceIndex` resource only needs the following actions.
```
"geo:DescribePlaceIndex",
"geo:ListPlaceIndexes",
"geo:GetPlace",
"geo:SearchPlaceIndexForPosition",
"geo:SearchPlaceIndexForSuggestions",
"geo:SearchPlaceIndexForText"
```

**Pull request closed without merging**